ddtrace/profiling/exporter/http.py

```diff
@@ -180,12 +158,20 @@ def export(self, events, start_time_ns, end_time_ns):
             # Retry after 1s, 2s, 4s, 8s with some randomness
             wait=tenacity.wait_random_exponential(multiplier=0.5),
             stop=tenacity.stop_after_delay(self.max_retry_delay),
-            retry=tenacity.retry_if_exception_type(
-                (error.HTTPError, error.URLError, http_client.HTTPException, OSError, IOError)
-            ),
+            retry=tenacity.retry_if_exception_type((http_client.HTTPException, OSError, IOError))
+            & tenacity.retry_if_exception(self._no_retry_on_404),
         )
 
         try:
             retry(request.urlopen, req, timeout=self.timeout)
         except tenacity.RetryError as e:
             raise UploadFailed(e.last_attempt.exception())
+        except error.HTTPError as e:
+            if e.code == 404 and not self.api_key:
+                LOG.warning(
+                    "Datadog agent does not support profiling. "
+                    "Configure agent-less profiling or upgrade your Datadog agent."
```

Comment:
Can we change the messaging here to be a bit clearer in what needs to be done and not mention agentless at all? Something along the lines of:

https://github.com/DataDog/dd-trace-go/blob/v1/profiler/upload.go#L26-L27

Reply:
Done.